src/main/java/com/hyperwallet/clientsdk/util/Message.java

```diff
@@ -0,0 +1,148 @@
+package com.hyperwallet.clientsdk.util;
+
+import cc.protea.util.http.Response;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+
+ Modified with modifications copyright Richard Stanford
+
+ -- Original portions by Joe Ernst --
+
+ Copyright 2012 Joe J. Ernst
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ */
+
+/**
+ * This class represents an HTTP Message, which could either be a Request or a Response. Message is an abstract class that contains fields and methods that are common to both types of Messages.
+ *
+ * @param <T>
+ *            A Type that extends Message (either {@link Request} or {@link Response})
+ */
+public abstract class Message<T extends Message<T>> {
```

Comment:
is this copied code?, can you make sure that we are not legally bounded with this approach? or can we also add something to recognize this persons work, please work with legal thanks.